src/compiler/interface.jl

```diff
@@ -50,9 +50,22 @@ sensitivity(y) = error("Output should be scalar; gradients are not defined for o
     gradient(f, args...)
 
 Returns a tuple containing `∂f/∂x` for each argument `x`,
-the derivative (for scalar x) or the gradient.
+the derivative (for scalar `x`) or the gradient.
 
 `f(args...)` must be a real number, see [`jacobian`](@ref) for array output.
+
+See also [`withgradient`](@ref) to keep the value `f(args...)`,
+and `pullback`](@ref) for value and back-propagator.
+
+```jldoctest; setup=:(using Zygote)
+julia> gradient(*, 2, 3, 5)
+(15, 10, 6)
+
+julia> gradient([7,11,13]) do x
+         sum(abs2, x)
+       end
+([14, 22, 26],)
```

Comment:
```suggestion
julia> gradient(x -> sum(abs2, x), [7,11,13])
([14, 22, 26],)
```

Reply:
just because the do syntax could be confusing for beginners

Reply:
True. Although it would also be good to teach them, since this is a common & useful idiom for gradients. I've added this plus a `do` variant, see what you think.